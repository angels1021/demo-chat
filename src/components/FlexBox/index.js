/**
 * FlexBox component
 *
 * @module components/FlexBox
 * @component FlexBox
 * @typicalname FlexBox
 *
 * @requires [React]{@link https://facebook.github.io/react/docs/hello-world.html}
 * @requires [prop-types]{@link https://www.npmjs.com/package/prop-types}
 * */
import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import Flex from '../../containers/Flex';

const FlexBox = ({
  align = '',
  justify = '',
  direction = '',
  className = '',
  children
}) => (
  <Flex {... {
    align,
    justify,
    direction,
    className
  }} >
    {(classes) => (<div className={cn(classes, 'd-flex')}>{children}</div>)}
  </Flex>
);

FlexBox.propTypes = {
  align: PropTypes.string, //eslint-disable-line react/require-default-props
  justify: PropTypes.string, //eslint-disable-line react/require-default-props
  direction: PropTypes.string, //eslint-disable-line react/require-default-props
  className: PropTypes.string, //eslint-disable-line react/require-default-props
  children: PropTypes.node.isRequired
};

export default FlexBox;
