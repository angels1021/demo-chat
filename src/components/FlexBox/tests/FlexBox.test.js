import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import FlexBox from '../index';

const childrenNode = <h1>hello</h1>;
describe('FlexBox Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FlexBox>{childrenNode}</FlexBox>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  const renderedComponent = shallow(
    <FlexBox
      align="center"
      justify="center"
      direction="column"
      className="otherClass andAnother"
    >
      {childrenNode}
    </FlexBox>
  ).dive();

  it('should return the match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should contain passed in children', () => {
    // assert
    expect(renderedComponent.contains(childrenNode)).toBe(true);
  });

  it('should add flex container classes', () => {
    expect(renderedComponent.hasClass('d-flex')).toBe(true);
  });

  it('should add correct align classes', () => {
    expect(renderedComponent.hasClass('align-center')).toBe(true);
    expect(renderedComponent.hasClass('justify-content-center')).toBe(true);
    expect(renderedComponent.hasClass('align-right')).toBe(false);
  });

  it('should have direction class matching the passed in direction prop', () => {
    // assert
    expect(renderedComponent.hasClass('flex-column')).toBe(true);
    expect(renderedComponent.hasClass('flex-row')).toBe(false);
  });

  it('should include any additional classes passed in', () => {
    // assert
    expect(renderedComponent.hasClass('otherClass')).toBe(true);
    expect(renderedComponent.hasClass('andAnother')).toBe(true);
  });
});
