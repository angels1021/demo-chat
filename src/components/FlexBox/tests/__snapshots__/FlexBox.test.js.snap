// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlexBox Component should return the match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Flex
    align="center"
    className="otherClass andAnother"
    direction="column"
    justify="center"
>
    [Function]
</Flex>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        hello
</h1>,
      "className": "align-center justify-content-center flex-column otherClass andAnother d-flex",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "hello",
      },
      "ref": null,
      "rendered": "hello",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          hello
</h1>,
        "className": "align-center justify-content-center flex-column otherClass andAnother d-flex",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
        },
        "ref": null,
        "rendered": "hello",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
